### PR TITLE
Saving and loading of obstacles

### PR DESCRIPTION
**Pull request by joshhting**
Partially addresses #86 and #74 
- [x] Fix visual formatting
- [x] Load and save specific filenames

### CODE REVIEW

File: src/rrt-viewer/RRTWidget.cpp [hunk 1]
```diff
@@ -60,6 +63,45 @@ void RRTWidget::reset() {
...
+    ifstream file("rrtconfig.txt");
+    ObstacleGrid& grid = _stateSpace->obstacleGrid();
+    char x;
+    if (file.is_open()) {
+        for (int j = 0; j < grid.discretizedHeight(); j++) {
+            for (int i = 0; i < grid.discretizedWidth(); i++) {
+                file >> x;
+                if ('1' == x) {
+                    grid.obstacleAt(i, j) = true;
+                } else if ('0' == x) {
```
Comment (jgkamat) [inline src/rrt-viewer/RRTWidget.cpp]: Can you use double quotes everywhere or single quotes everywhere :P

Comment (joshhting) [reply src/rrt-viewer/RRTWidget.cpp]: True

**Comment by joshhting** [comment]
Do you know how to fix the margin things on the rrt viewer that this commit messed up

**Comment by chachmu** [comment]
My screen is a weird resolution so @jgkamat is going to fix it for normal people's screens.

**Comment by joshhting** [comment]
The gui is a little off sync from the margin change, do you know how to call the qml properties from [this line](https://github.com/RoboJackets/rrt/blob/master/src/rrt-viewer/RRTWidget.cpp#L13)

**Comment by chachmu** [comment]
@joshhting I think you can call setMinimumSize(200, 200) for example but I don't know how that interacts with the seperate QML stylesheet, it may not actually change anything.

**Comment by joshhting** [comment]
I mean do you know how to change the hardcoded values in the line I linked above to point to whatever the QML's values are

**Comment by chachmu** [comment]
@joshhting have you tried size()? https://doc.qt.io/qt-5.10/qwidget.html#size-prop

**Comment by joshhting** [comment]
Is the rrt still on qt4? The RRTWidget doesn't seem to have a size()...

**Comment by jgkamat** [comment]
RRTWidget is a subclass of [QQuickPaintedItem](https://doc.qt.io/qt-5/qquickpainteditem.html), which you can use 'width' and 'height' on. It dosen't seem to be available right away in the constructor though, but it seems to be correct in the paint method.

**Comment by joshhting** [comment]
Yeah I was confused as to why it wasn't working in the constructor

**Comment by joshhting** [comment]
Working around the bug/feature where the dimensions are not available in the constructor seems too complicated for my level of Qt experience, so I'll leave it as a low priority issue for a more capable person.  We could make it so that the part of the constructor that requires the width & height is called in some separate setupRRT() function that is called after the RRTWidget is initialized so that width() and height() work as expected, but I'm not sure how to call the RRTWidget from the QMLApplicationEngine.

**Comment by joshhting** [comment]
I'm going to merge this in on Wednesday if there are no complaints

**Comment by jgkamat** [comment]
Yay, :D

**Pull request merged**